library/datasort.h

```diff
@@ -81,8 +81,10 @@ struct datasort_cfg {
 	struct list_head		sorted_chunks;
 	/* Result of mergesort */
 	struct datasort_chunk		*result;
-	/* Datasort directory */
-	char				*dir;
+	/* Datasort directory - chunks are merged into single sorted blob here */
+	char				*datasort_dir;
+	/* Chunks directory - if not defined, then chunks are stored in datasort_dir */
+	char				*chunks_dir;
```

Comment:
This is not the answer at all. Even besides the fact, that you haven't answered why there are **2** directories.
Please make your homework describing the purpose of your changes in more details.

Currently datasort and defrag machinery uses data directory as a base. I propose (and abovementioned #121 issue) to change that base dir to special dedicated directory specified in config, which by default will be equal to data directory. This is much smaller and cleaner patch as well as it doesn't need to introduce 2 directories. In particular, this hunk above will not be changed at all, defrag setup will set `dir` to config datasort dir + suffix